src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,119 @@
+        elevatorMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetElevatorMeters, 0),
+                new TrapezoidProfile.State(getElevatorPositionMeters(), getElevatorVelocityInMetersPerSeconds())
+        );
+    }
+
+    void generateAngleMotorProfile(Rotation2d targetAngle) {
+        angleMotorProfile = new TrapezoidProfile(
+                ArmConstants.ANGLE_CONSTRAINTS,
+                new TrapezoidProfile.State(targetAngle.getDegrees(), 0),
+                new TrapezoidProfile.State(getAnglePosition().getDegrees(), getAngleVelocityInDegreesPerSeconds())
+        );
+
+        lastAngleMotorProfileGeneration = Timer.getFPGATimestamp();
+    }
+
+    private Rotation2d getAnglePosition() {
+        double positionRevolutions = ArmConstants.ANGLE_ENCODER_POSITION_SIGNAL.refresh().getValue();
+        return Rotation2d.fromRotations(positionRevolutions);
+    }
+
+    private double getElevatorPositionMeters() {
+        double elevatorPositionMagTicks = elevatorEncoder.getSelectedSensorPosition();
+        double elevatorPositionRevolution = Conversions.magTicksToRevolutions(elevatorPositionMagTicks);
+        return elevatorPositionRevolution / ArmConstants.ELEVATOR_METERS_PER_REVOLUTION;
+    }
+
+    private double getAngleVelocityInDegreesPerSeconds() {
+        return Conversions.revolutionsToDegrees(ArmConstants.ANGLE_ENCODER_VELOCITY_SIGNAL.refresh().getValue());
+    }
+
+    private double getElevatorVelocityInMetersPerSeconds(){
+        double elevatorVelocityMagTicksPer100ms = elevatorEncoder.getSelectedSensorVelocity();
+        double elevatorVelocityMagTicksPerSecond = Conversions.perHundredMsToPerSecond(elevatorVelocityMagTicksPer100ms);
+        double elevatorVelocityRevolutionPerSecond = Conversions.magTicksToRevolutions(elevatorVelocityMagTicksPerSecond);
+        return elevatorVelocityRevolutionPerSecond / ArmConstants.ELEVATOR_METERS_PER_REVOLUTION;
+    }
+
+    void setTargetAngleFromProfile() {
+        if (angleMotorProfile == null) {
+            masterAngleMotor.stopMotor();
+            followerAngleMotor.stopMotor();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = angleMotorProfile.calculate(getAngleProfileTime());
+        double output = ArmConstants.ANGLE_PID_CONTROLLER.calculate(getAnglePosition().getDegrees(), targetState.position);
+        masterAngleMotor.setVoltage(output);
+        followerAngleMotor.setVoltage(output);
+    }
+
+    void setTargetElevatorPositionFromProfile() {
+        if (angleMotorProfile == null) {
+            masterElevatorMotor.stopMotor();
+            followerElevatorMotor.stopMotor();
+            return;
+        }
+
+        TrapezoidProfile.State targetState = elevatorMotorProfile.calculate(getElevatorProfileTime());
+        double output = ArmConstants.ELEVATOR_PID_CONTROLLER.calculate(getElevatorPositionMeters(), targetState.position);
```

Comment:
calculate feedforward out here from the ElevatorFeedforward class (explained in the motion profiles presentation)